### ISSUE TITLE
`Stack`/`Inline`/`Tiles` - does not apply styles to non React elements

### ISSUE THREAD

**Issue by victorouse** [jameslnewell/substance#30]
## Description 

Since `React.isValidElement` is called on children passed to `Stack`/`Inline`/`Tiles` to wrap them in their respective `Item` components, styling does not apply to elements created without `React.createElement`, i.e. a raw text node

- [Inline](https://github.com/jameslnewell/substance/blob/1f93498cfebb3dda80d5284d27f1f817f181937e/packages/layout/src/Inline/index.tsx#L131)
- [Stack](https://github.com/jameslnewell/substance/blob/1f93498cfebb3dda80d5284d27f1f817f181937e/packages/layout/src/Stack/index.tsx#L80)
- [Tiles](https://github.com/jameslnewell/substance/blob/1f93498cfebb3dda80d5284d27f1f817f181937e/packages/layout/src/Tiles/index.tsx#L146)

## Example

```tsx
const Foo = () => (
  <Inline spaceX="xs">
    <h1>Foo</h2>
    bar
  </Inline>
)
```

**Comment by jameslnewell** [comment]
What should this produce?

```js
const Foo = () => (
  <Inline spaceX="xs">
    foo
    bar
  </Inline>
)
```

**Comment by victorouse** [comment]
> What should this produce?
> 
> ```js
> const Foo = () => (
>   <Inline spaceX="xs">
>     foo
>     bar
>   </Inline>
> )
> ```
> 
> A single child or two?

Yeah, interesting case: not sure if this is a contrived example since I'm using `grid`, but `Inline` feels a bit like `grid` with `grid-template-columns: repeat(auto-fit, [spaceX])`:

![image](https://user-images.githubusercontent.com/1930296/98300155-fd854c80-200c-11eb-9958-dad2d97fc0b3.png)

Which looks like _if_ there is a DOM node present, space is created between it and the text node:

![image](https://user-images.githubusercontent.com/1930296/98300692-c19eb700-200d-11eb-805e-7e7f8ad5922d.png)

But in the case where it's just text nodes, it's treated as a single "item".

This _might_ echo the original use-case of:

```js
const Foo = () => (
  <Inline spaceX="xs">
    <h1>Foo</h2>
    bar
  </Inline>
)
```

https://flems.io/#0=N4IgtglgJlA2CmIBcB2FA6ArAGhAZ3gQGMAXeKZEdACxLFhFwDMIE9kBtUAOwEMxESKrXqMQRAPbcy0ygB48JAJ4IAfAB1uAAi3oI3WPvhbgmnTqgQ8AB1i8lSLQHMATtADcZ86+gBaMmC2vGS+krAArmDceI4u8NbwwQAUvOEkEr4sJNhaAEyY1gAeAJSe2t5uUKESEVG+TrzWjrkADEVl5lpgvC5O+r4ARhIk6WDNBYUd5kMuUPAujgCMRVpQvHjU5FoDdkQA1lMAvl5e6EOFJl46mxBOtM1tk1daAO7QJNQP7V7H5ac74WMpnKOgGvH2rgk4W4UEcAPgRxO5XQcSglxB23Be0h0NhWlRiL+yNc8Hg2mBnTBEJcUJhjhJZMJcgA9IoVPANNxNJo5JYAG5aIh2PB4AC86hA+kM3HgEs5Ol5EAFQvWYol522EVlIFULP58q0iuVwrVIA1qLleqVBqNgpN4rNEguDK5OqtfM57s5PP5dtVDqlRjlXiYEgkXltKpFDvN5EtzP1Ed9UdNGpd8cT3C93KzyftEsDMuD5VD4fKYJcSaVfuj6qdzjiZIz1p5CZbXNz1ZTAYMQZ1IbDXgrQ94AC9W-qxARiCQIFJ2EJFoskIsAGwgQ7YHj8QRUABW7FwkmkZJIlA3W5AfAElHQRBFYmPMjPQg3AF1DkA
